src/rules/no-useless-path-segments.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @fileOverview Ensures that there are no useless path segments
+ * @author Thomas Grainger
+ */
+
+import path from 'path'
+import sumBy from 'lodash.sumby'
+import escapeRegExp from 'lodash.escaperegexp'
+import resolve from 'eslint-module-utils/resolve'
+import moduleVisitor from 'eslint-module-utils/moduleVisitor'
+
+function toRel(rel, sep) {
+  const rSep = escapeRegExp(sep)
```

Comment:
is there a way this could be done without having to escape regexes, or use them at all?